src/object.jl

```diff
@@ -144,6 +144,46 @@ function _download_task(io; kw...)
     return nothing
 end
 
+_ndownload_tasks(total_size, part_size, numthreads=Threads.nthreads()) =
+    min(numthreads, max(1, div(total_size, part_size, RoundUp)))
+
+"""
+    PrefetchedDownloadStream{T <: Object} <: IO
+
+A buffered, read-only, in-memory IO stream that fetches chunks from remote cloud `Object`.
+
+`prefetch_multipart_size` is the max size of any individual GET request in bytes
+(default $(Base.format_bytes(DEFAULT_PREFETCH_MULTIPART_SIZE))),
+`prefetch_size` is the size of a buffer that stores the fetched bytes and which is iterated
+when we consume/read the IO (default $(Base.format_bytes(DEFAULT_PREFETCH_SIZE))).
+
+Once you start reading from the prefetch buffer, a secondary buffer will began to be populated
+by multiple background download tasks (meaning that this object allocates `2*prefetch_size` bytes
+for buffers).
+These background tasks are `@spawn`ed when this object is instantiated and by default 4 tasks
+(~ `prefetch_size` / `prefetch_multipart_size`) are spawned for performing the GET requests +
+1 task is spawned to coordinate the prefetching process. Number of spawned tasks depends on the
+size of the input and the number of threads available (see [`_ndownload_tasks`](@ref) helper function).
```

Comment:
What is ``(~ `prefetch_size` / `prefetch_multipart_size`)`` referring to here?  The size of the buffer for each task? the number of tasks? I don't follow, since it seems like the number of tasks is size of input / number of threads?

Reply:
>4 tasks (~ `prefetch_size` / `prefetch_multipart_size`) are spawned

The number of tasks, as an approximation. The second sentence perhaps should just say the the number of tasks is bounded by the number of threads.

Reply:
Ah, yeah, that'd be helpful. Thanks! Maybe expand both a bit? Maybe:
```
These background tasks are `@spawn`ed when this object is instantiated, with approximately
`prefetch_size` / `prefetch_multipart_size` tasks spawned for performing the GET requests
(defaults to 4 if those fields aren't specified) + 1 task is spawned to coordinate the prefetching
process. Number of spawned tasks is upper-bounded by the
size of the input and the number of threads available (see [`_ndownload_tasks`](@ref) helper function).
```
Or something like that?